common/lib/capa/capa/capa_problem.py

```diff
@@ -855,17 +870,76 @@ def _preprocess_problem(self, tree):  # private
                 id=response_id_str
             )
 
-            # assign one answer_id for each input type or solution type
+            # assign one answer_id for each input type
             for entry in inputfields:
                 entry.attrib['response_id'] = str(response_id)
                 entry.attrib['answer_id'] = str(answer_id)
                 entry.attrib['id'] = "%s_%i_%i" % (self.problem_id, response_id, answer_id)
                 answer_id = answer_id + 1
 
-            # Find the label and save it for html transformation step
-            responsetype_label = response.find('label')
-            problem_data[self.problem_id + '_' + str(response_id)] = {
-                'label': responsetype_label.text if responsetype_label is not None else ''
+            question_id = self.problem_id + '_' + str(response_id)
+            label = ''
+            element_to_be_deleted = None
+
+            # Extract label value from <label> tag or label attribute from inside the responsetype
+            responsetype_label_tag = response.find('label')
+            if responsetype_label_tag is not None:
+                label = responsetype_label_tag.text
+                # Delete <label> tag containing question text otherwise question will be rendered twice
+                element_to_be_deleted = responsetype_label_tag
+            elif 'label' in inputfields[0].attrib:
+                # Extract label value from label attribute
+                # This is the case when we have a problem
+                # * with multiple questions without separation
+                # * single question with old XML format only
+
+                label = inputfields[0].attrib['label']
+                # Get first <p> tag before responsetype
+                p_tag = response.xpath('preceding-sibling::p[1]')
+
+                if p_tag:
+                    # It may be possible that label attribute value doesn't match with <p> tag
+                    # This happens when author updated the question <p> tag directly in XML but
+                    # didn't changed the label attribute value. In this case we will consider the
+                    # first <p> tag before responsetype as question.
+                    if label != p_tag[0].text:
+                        label = p_tag[0].text
+                    element_to_be_deleted = p_tag[0]
+            else:
+                # neither <label> tag nor label attribute is present inside responsetype
+                # This is the case when we have a problem with multi-questions without --- having markdown
+                # find the immediate <label> tag before the responsetype. also delete it avoid rendering twice
```

Comment:
Can we update these comments with the following text

> In this case the problems don't have <label> tag or label attribute inside the responsetype so we will get the first preceding label tag w.r.t to this responsetype.
> This will take care of those multi-question problems that are not using `---` in their markdown.